src/rpc/misc.cpp

```diff
@@ -449,6 +451,36 @@ static UniValue getmemoryinfo(const JSONRPCRequest& request)
     }
 }
 
+static UniValue getgeneralinfo(const JSONRPCRequest& request)
+{
+    RPCHelpMan{"getgeneralnfo",
+                "\nReturns data about the bitcoin daemon.\n",
+                {},
+                RPCResult{
+                "  {\n"
+                "    \"clientversion\": \"version string\",     (string) Client version\n"
+                "    \"useragent\":\"client name\",             (string) Client name\n"
+                "    \"datadir\":\"datadir path\",              (string) Data directory path\n"
+                "    \"blocksdir\":\"blocksdir path\",          (string) Blocks directory path\n"
+                "    \"startuptime\":\"string\",                (string) Startup time\n"
+                "  }\n"
+                },
+                RPCExamples{
+                    HelpExampleCli("getgeneralinfo", "")
+            + HelpExampleRpc("getgeneralinfo", "")
+                },
+            }.Check(request);
+
+        UniValue obj(UniValue::VOBJ);
+        obj.pushKV("clientversion", FormatFullVersion());
+        obj.pushKV("useragent", strSubVersion);
+        obj.pushKV("datadir", GetDataDir().string());
+        obj.pushKV("blocksdir", GetBlocksDir().string());
+        obj.pushKV("startuptime", FormatISO8601DateTime(GetStartupTime()));
+        return obj;
+
+}
```

Comment:
Is it better to move this function to [`server.cpp`](https://github.com/bitcoin/bitcoin/blob/a654626f076a72416a3d354218d7107571d6caaf/src/rpc/server.cpp) since that file contains the majority of `control` commands, like `getrpcinfo` and `uptime`?

Reply:
I could move them there. Actually, I put them in misc.cpp because I wasn't sure where to put it best. Maybe we should wait for additional input from others before changing the current variant?